### DISCUSSION TITLE
Different behavior when retrieving data from a bucket using jsonpath in 9.0 (#each)?

### DISCUSSION THREAD

**Discussion by NicolasCelesta** [mockoon/mockoon#1582] [Help] [answered]
Hi,

We are currently using V8.4 and trying to upgrade to V9.0. However, we're facing a challenge as practically most of our mocks are returning empty responses in V9.0.

Investigating, it seems jsonpath doesn't work as it used to (?) on our environment.
Typically, the below doesn't select anything anymore for some reason (Comprehensive example down below).
````
$[?(@.status==='enabled' && (@.id===1 || @.id===2))]
````

while this does (Still)
````
$[?(@.status==='enabled' && @.id===1)]
````

We also noticed this is working as we would expect when using the [online tool](https://mockoon.com/tools/json-object-path-evaluator/), it's just not working on our local environment (Tested on Windows).

Since there is nothing in the release note nor anyone commenting on that issue here, I'm guessing we're doing something incorrectly, but can actually figure out what. 

Any idea what we're doing wrong?

Thanks you very much for your help 🙏!

-------
Assuming the following sample in a data bucket 'data-bucket'.
```json
[
   {
      "id":1,
      "status":"enabled"
   },
   {
      "id":2,
      "status":"enabled"
   },
   {
      "id":3,
      "status":"disabled"
   }
]
```

The following template ('/ok' in the [enclosed environment](https://github.com/user-attachments/files/17719818/test-path.json)):
```
[
  {{#each (dataRaw 'data-bucket' "$[?(@.status==='enabled' && @.id===1)]")}}
   {
      "id": {{this.id}},
      "status":"{{this.status}}"
   }
  {{/each}}
]
```

Would return
```json
[
    {
        "id": 1,
        "status": "enabled"
    }
]
```

However, the following template ('/ko' in the [enclosed environment](https://github.com/user-attachments/files/17719818/test-path.json)):
```
[
  {{#each (dataRaw 'data-bucket' "$[?(@.status==='enabled' && (@.id===1 || @.id===2))]")}}
   {
      "id": {{this.id}},
      "status":"{{this.status}}"
   }
  {{/each}}
]
```

Would return
```json
[]
```

**Comment by 255kb** [comment, marked as answer]
Hi, sorry if this impacted you. 
A [change](https://github.com/mockoon/mockoon/issues/1551) was made in v9 to prevent RCE vulnerabilities linked to JSONPath. 
We went a step further and decided to whitelist some filter syntaxes using a regex. It was in the list of changes, but not listed as a potential breaking change as we thought we covered most cases. I will update the release note to make it clearer it's potentially breaking. 

The problematic syntax here is the enclosed brackets `( ... || ... )`. I will open an issue and include these syntaxes for the next release (end of November). 
As a temporary workaround you can get rid of the brackets as the OR has the priority anyway: 
`$[?(@.status==='enabled' && @.id===1 || @.id===2)]` 

Update: [issue](https://github.com/mockoon/mockoon/issues/1583)

**Comment by NicolasCelesta** [comment]
Hi,

Ha ok, understood. We'll be waiting for that update then.

Thanks you for your help!

Nicolas
